src/goto-programs/goto_convert.cpp

```diff
@@ -1830,8 +1830,11 @@ bool goto_convertt::get_string_constant(
       forall_operands(it, index_op)
         if(it->is_constant())
         {
-          unsigned long i = integer2ulong(
-            bv2integer(id2string(to_constant_expr(*it).get_value()), true));
+          mp_integer int_value;
+          if(to_integer(to_constant_expr(*it), int_value))
+            return true;
+
+          unsigned long i = integer2ulong(int_value);
```

Comment:
```suggestion
          const auto i = numeric_cast<std::size_t>(*it);
          if(!i.has_value())
            return true;
```
and then use `*i` below.

Reply:
Say here we add a 'const'.

Reply:
I've tried to explain my thinking above, maybe that helps?